inspirehep/base/views.py

```diff
@@ -158,17 +156,9 @@ def data():
                  'collection': (unicode, "")})
 def ajax_references(recid, collection):
     """Handler for datatables references view"""
-    from flask import jsonify
-    from inspirehep.utils.references import Reference
-    from invenio.base.globals import cfg
-    from invenio_ext.es import es
 
-    index = cfg['SEARCH_ELASTIC_COLLECTION_INDEX_MAPPING'].get(collection, 'hep')
-    return jsonify(
-        {
-            "data": Reference(es.get(index=index, id=recid)['_source']).references()
-        }
-    )
+    from inspirehep.utils.references import AjaxReferences
+    return AjaxReferences.get_references(unicode(recid), unicode(collection.lower()))
```

Comment:
This is passed directly to ES. It should be able to cast on its own an int to a unicode.